### ISSUE TITLE
requires_pylab decorator is wrong

### ISSUE THREAD

**Issue by DougBurke** [sherpa/sherpa#1971]
I added this test to `sherpa/ui/tests/test_ui_plot.py`:

```
@requires_pylab
def test_this_should_fail():
    assert False
```

and then try to run it with

```
% pytest sherpa/ui/tests/test_ui_plot.py::test_this_should_fail
pytest-xvfb could not find Xvfb. You can install it to prevent windows from being shown.
============================================== test session starts ===============================================
platform linux -- Python 3.11.4, pytest-7.4.0, pluggy-1.2.0
rootdir: /home/dburke/sherpa/sherpa-xspec
configfile: pytest.ini
plugins: xvfb-3.0.0, doctestplus-1.0.0, anyio-4.0.0, cov-4.1.0, xdist-3.5.0
collected 0 items                                                                                                

================================================ warnings summary ================================================
../../micromamba/envs/sherpa-xspec/lib/python3.11/site-packages/_pytest/mark/structures.py:350
  /home/dburke/micromamba/envs/sherpa-xspec/lib/python3.11/site-packages/_pytest/mark/structures.py:350: PytestCollectionWarning: cannot collect 'test_plot_xxx_components_simple' because it is not a function.
    def __call__(self, *args: object, **kwargs: object):

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=============================================== 1 warning in 0.47s ===============================================
ERROR: not found: /home/dburke/sherpa/sherpa-xspec/sherpa/ui/tests/test_ui_plot.py::test_this_should_fail
(no name '/home/dburke/sherpa/sherpa-xspec/sherpa/ui/tests/test_ui_plot.py::test_this_should_fail' in any of [<DocTestModulePlus sherpa/ui/tests/test_ui_plot.py>, <Module sherpa/ui/tests/test_ui_plot.py>])

```

If you remove the `@requires_pylab` line then the test is run (and fails, as expected).

The problem actually is reported by `PytestCollectionWarning`: it doesn't think this is a function. This is because `requires_pylab` is written to be a decorator but actually behaves more like a pytest fixture:

```
   def requires_pylab(test_function):
        """Runs the test with the PylabBackend plotting backend if available,
        otherwise skips the test.
        """
        pylab_backend = pytest.importorskip("sherpa.plot.pylab_backend")
        plt = pytest.importorskip("matplotlib.pyplot")

        with TemporaryPlottingBackend(pylab_backend.PylabBackend()):
            yield

        plt.close(fig="all")
```

Note that it does nothing with `test_function`!

There's actually other things we could look at here too:

- `sherpa.utils.testing` allows `pytest` to be optional, but really it is required for running tests, but we don't want to require `pytest` jus tto use sherpa
- all the `requires_xxx` routines in this module should probably be moved to `sherpa/conftest.py`, which might simplify this

**Comment by DougBurke** [comment]
And "fixing" `requires_pylab` is going to show us a bunch of tests that currently fail.

**Comment by DougBurke** [comment]
So, we see a test failing because (once this is fixed) matplotlib 3.8.x (just recently released) behaves differently for contour plots and it triggers a failure. Joy.
